LoopKit/InsulinKit/InsulinDeliveryStore.swift

```diff
@@ -69,16 +71,35 @@ public class InsulinDeliveryStore: HealthKitSampleStore {
         )
 
         cacheStore.onReady { (error) in
-            // Should we do something here?
+            cacheStore.fetchAnchor(key: InsulinDeliveryStore.queryAnchorMetadataKey) { (anchor) in
+                self.queue.async {
+                    self.queryAnchor = anchor
+
+                    if !self.authorizationRequired {
+                        self.createQuery()
+                    }
+                }
+            }
         }
     }
+    
+    // MARK: - HealthKitSampleStore
 
-    public override func processResults(from query: HKAnchoredObjectQuery, added: [HKSample], deleted: [HKDeletedObject], error: Error?) {
-        guard error == nil else {
-            return
-        }
+    override func queryAnchorDidChange() {
+        cacheStore.storeAnchor(queryAnchor, key: InsulinDeliveryStore.queryAnchorMetadataKey)
+    }
 
+    override func processResults(from query: HKAnchoredObjectQuery, added: [HKSample], deleted: [HKDeletedObject], anchor: HKQueryAnchor, completion: @escaping (Bool) -> Void) {
         queue.async {
+            guard anchor != self.queryAnchor else {
+                self.log.default("Skipping processing results from anchored object query, as anchor was already processed")
+                if self.lastBasalEndDate == nil {
```

Comment:
Why do we need this check here? In this path nothing is changing in the cache so we shouldn't need to update the last basal date, right?

Reply:
I see that we do it below even if the cache hasn't changed. Not sure why there, either.

Reply:
There was an issue where last basal end date was never set up properly after app restart, and we do get an initial query response from HK every time, which is why this was set here. We could probably move it into the actual initialization of InsulinDeliveryStore; I'll look into that.